sdk/storage/azure-storage-blobs/test/ut/append_blob_client_test.cpp

```diff
@@ -7,43 +7,30 @@
 
 namespace Azure { namespace Storage { namespace Test {
 
-  std::shared_ptr<Azure::Storage::Blobs::AppendBlobClient> AppendBlobClientTest::m_appendBlobClient;
-  std::string AppendBlobClientTest::m_blobName;
-  Azure::Storage::Blobs::CreateAppendBlobOptions AppendBlobClientTest::m_blobUploadOptions;
-  std::vector<uint8_t> AppendBlobClientTest::m_blobContent;
+  void AppendBlobClientTest::SetUp() { BlobContainerClientTest::SetUp(); }
 
-  void AppendBlobClientTest::SetUpTestSuite()
+  void AppendBlobClientTest::TearDown()
   {
-    BlobContainerClientTest::SetUpTestSuite();
-
-    m_blobName = RandomString();
-    auto appendBlobClient = Azure::Storage::Blobs::AppendBlobClient::CreateFromConnectionString(
-        StandardStorageConnectionString(), m_containerName, m_blobName);
-    m_appendBlobClient
-        = std::make_shared<Azure::Storage::Blobs::AppendBlobClient>(std::move(appendBlobClient));
-    m_blobContent.resize(100);
-    RandomBuffer(reinterpret_cast<char*>(&m_blobContent[0]), m_blobContent.size());
-    m_blobUploadOptions.Metadata = {{"key1", "V1"}, {"key2", "Value2"}};
-    m_blobUploadOptions.HttpHeaders.ContentType = "application/x-binary";
-    m_blobUploadOptions.HttpHeaders.ContentLanguage = "en-US";
-    m_blobUploadOptions.HttpHeaders.ContentDisposition = "attachment";
-    m_blobUploadOptions.HttpHeaders.CacheControl = "no-cache";
-    m_blobUploadOptions.HttpHeaders.ContentEncoding = "identify";
-    m_blobUploadOptions.HttpHeaders.ContentHash.Value.clear();
-    m_appendBlobClient->Create(m_blobUploadOptions);
-    auto blockContent
-        = Azure::Core::IO::MemoryBodyStream(m_blobContent.data(), m_blobContent.size());
-    m_appendBlobClient->AppendBlock(blockContent);
-    m_blobUploadOptions.HttpHeaders.ContentHash
-        = m_appendBlobClient->GetProperties().Value.HttpHeaders.ContentHash;
+    if (m_appendBlobClient)
+    {
+      m_appendBlobClient->Delete();
+    }
+    // Delete container
+    BlobContainerClientTest::TearDown();
   }
 
-  void AppendBlobClientTest::TearDownTestSuite() { BlobContainerClientTest::TearDownTestSuite(); }
-
+  // Requires blob versioning?
   TEST_F(AppendBlobClientTest, CreateAppendDelete)
   {
-    auto appendBlobClient = Azure::Storage::Blobs::AppendBlobClient::CreateFromConnectionString(
-        StandardStorageConnectionString(), m_containerName, RandomString());
+
+    auto const testName(GetTestName());
```

Comment:
It makes it obscure to use `auto` keyword here. I cannot figure out what the type is at the first glance. How about 

```C++
const std::string testName = GetTestName();
```

Reply:
`const std::string testName(GetTestName()); `  that's how I did other places.